sdcm/utils/version_utils.py

```
@@ -281,15 +281,27 @@ def parse(version_string: str):
+    callers can distinguish a permanently-unavailable resource (eg. archived to cold
+    storage, returns 403/404) from other, less expected failures."""
+
+    def __init__(self, message: str, status_code: int = None):
+        super().__init__(message)
+        self.status_code = status_code
+
+
 @lru_cache(maxsize=1024)
 @retrying(n=10, sleep_time=0.1)
 def get_url_content(url, return_url_data=True):
     response = requests.get(url=url)
```

Ruff (0.15.21)

[error] Probable use of `requests` call without timeout

(S113)